batch/Makefile

```diff
@@ -40,14 +40,31 @@ deploy: build
 	kubectl -n $(NAMESPACE) apply -f service-account.yaml.out
 	kubectl -n $(NAMESPACE) apply -f deployment.yaml.out
 
-.PHONY: create-build-worker-image-instance
-create-build-worker-image-instance:
+.PHONY: gcp-create-build-worker-image-instance
+gcp-create-build-worker-image-instance:
 	-gcloud -q compute --project $(PROJECT) instances delete --zone=$(ZONE) build-batch-worker-image
-	python3 ../ci/jinja2_render.py '{"global":{"docker_root_image":"$(DOCKER_ROOT_IMAGE)"}}' build-batch-worker-image-startup.sh build-batch-worker-image-startup.sh.out
-	gcloud -q compute --project $(PROJECT) instances create --zone=$(ZONE) build-batch-worker-image --machine-type=n1-standard-1 --network=default --network-tier=PREMIUM --metadata-from-file startup-script=build-batch-worker-image-startup.sh.out --no-restart-on-failure --maintenance-policy=MIGRATE --scopes=https://www.googleapis.com/auth/cloud-platform --image=$$(gcloud compute images list --standard-images --filter 'family="ubuntu-minimal-2004-lts"' --format='value(name)') --image-project=ubuntu-os-cloud --boot-disk-size=10GB --boot-disk-type=pd-ssd
+	python3 ../ci/jinja2_render.py '{"global":{"docker_root_image":"$(DOCKER_ROOT_IMAGE)"}}' build-batch-worker-image-startup-gcp.sh build-batch-worker-image-startup-gcp.sh.out
+	gcloud -q compute --project $(PROJECT) instances create --zone=$(ZONE) build-batch-worker-image --machine-type=n1-standard-1 --network=default --network-tier=PREMIUM --metadata-from-file startup-script=build-batch-worker-image-startup-gcp.sh.out --no-restart-on-failure --maintenance-policy=MIGRATE --scopes=https://www.googleapis.com/auth/cloud-platform --image=$$(gcloud compute images list --standard-images --filter 'family="ubuntu-minimal-2004-lts"' --format='value(name)') --image-project=ubuntu-os-cloud --boot-disk-size=10GB --boot-disk-type=pd-ssd
 
-.PHONY: create-worker-image
-create-worker-image:
+.PHONY: gcp-create-worker-image
+gcp-create-worker-image:
 	-gcloud -q compute --project $(PROJECT) images delete batch-worker-12
 	gcloud -q compute --project $(PROJECT) images create batch-worker-12 --source-disk=build-batch-worker-image --source-disk-zone=$(ZONE)
 	gcloud -q compute --project $(PROJECT) instances delete --zone=$(ZONE) build-batch-worker-image
+
+.PHONY: az-create-build-worker-image-instance
+az-create-build-worker-image-instance:
+	REGION=$(REGION) \
+	  RESOURCE_GROUP=$(RESOURCE_GROUP) \
+	  CONTAINER_REGISTRY_NAME=$(CONTAINER_REGISTRY_NAME) \
+	  DOCKER_ROOT_IMAGE=$(DOCKER_ROOT_IMAGE) \
+	  sh az-create-build-worker-image-instance.sh
+	# Need to SSH into the VM and run "sudo waagent -deprovision+user -force"
```

Comment:
Can you explain why this is necessary? Also could it just be done in the shell script with `run-command`?

Reply:
I tried to automate it, but the command just hangs and will never finish. Works in less than a second once you login to the node.